### PR TITLE
Savefig function, better 2D hist and utility documentation

### PR DESCRIPTION
**Pull request by 0ctagon**
I open this pull request to have a base for discussion about a function to save plots with correct figsize, subplots positions and dimensions.
The function provided here works wonderfully for any 1d or comparisons plots. It doesn't work at all for 2D plots (the colorbar is always out).
We will learn a lot about matplotlib limits, whether this PR will be merged or not.

### CODE REVIEW

**Comment by 0ctagon** [comment]
After further investigation, it doesn't work with 2D plots because the figsize is too small. We specify a squared figsize to get a square 2D hist, but the final figsize has to be larger to at least display the label of the colorbar.

**Comment by 0ctagon** [comment]
https://plothist.readthedocs.io/en/savefig/usage/utilities.html

**Comment by cyrraz** [comment]
Let's discuss it, I would not merge this function if it is not compatible with all the currently available figures.

**Comment by 0ctagon** [comment]
Now, everything works with the custom `savefig`. For 2D plots, `adjust_after_tight_layout` has to be used to adjust the space between `ax_colorbar` and `ax`.

**Comment by 0ctagon** [comment]
https://plothist.readthedocs.io/en/savefig/usage/utilities.html#save-figure gif added :smile:

File: plothist/plothist_style.py
Comment (cyrraz) [inline plothist/plothist_style.py:254]: `type(x) not in [float, int]`

File: plothist/plothist_style.py
Comment (cyrraz) [inline plothist/plothist_style.py:261]: `type(y) not in [float, int]`

File: plothist/plotters.py
Comment (cyrraz) [inline plothist/plotters.py:321]: None -> none

File: plothist/plotters.py
Comment (cyrraz) [inline plothist/plotters.py:831]: why not always doing it?

**Review by cyrraz — CHANGES_REQUESTED** [review]
Just a few comments, but nearly ready to be merged

File: plothist/plotters.py
Comment (0ctagon) [reply plothist/plotters.py:831]: Always doing it can mess up some plots. Not doing it cannot mess up plots, only make some subplots misaligned on a very few examples.

**Review by cyrraz — APPROVED** [review]

**Pull request merged**
